### ISSUE TITLE
.cody/ignore

### ISSUE THREAD

**Issue by abeatrix** [sourcegraph/cody#2756]
Introduced in https://github.com/sourcegraph/cody/pull/1382

Currently behind the `cody.internal.unstable` configuration.

Once we have gathered extensive internal feedback, we should move it to `experimental` stage and enable it by default for external input.

- [x] Work in a single workspace
- [x] Work in non-indexed workspace
- [x] Work in multi-root workspace
- [x] Work with nested repositories
- [x] Work with Chat
- [x] Work with Commands  
- [x] Work with Edits
- [x] Work with Autocompletes
- [x] Work on Mac
- [ ] Work on Windows 

## To be discussed

Do we need it to work with the following?
- [ ] Work with local embeddings during indexing stage
- [ ] Work with remote embeddings during indexing stage
- [ ] Work with symf during indexing stage

## Follow up
- UX 
  - how to let user know that their files were ignored (if needed)
    - [ ] in chat
    - [ ] in commands
    - [ ] in autocompletes
  - how can users find out about this feature
- [ ] Doc update

**Comment by abeatrix** [comment]
Question for you @DanTup 🙂 
Can the windows test we just merged help us with confirming/checking if the .cody/ignore file is working on Windows?

**Comment by DanTup** [comment]
If there are e2e tests, it should (the other tests were already running on Windows).

I can manually test things on Windows too if you want some extra validation (let met know if there are specifics things you want tested).

**Comment by DanTup** [comment]
We do have a context test here that validates ignores that works correctly on Windows:

https://github.com/sourcegraph/cody/blob/4fb31b80f5a1d6fcaa661d22d9911f2133428cef/vscode/src/editor/utils/editor-context.test.ts#L64

I don't know if it covers all the places where files could be returned though. It's checking that `getFileContextFiles()` is filtering out the ignored files returned in `vscode.workspace.findFile` but it might not be verifying if those ignored files show up from other context sources.

**Comment by abeatrix** [comment]
@DanTup thanks for the info. I'll create an issue to add an e2e test for Cody/ignore.
Will also create a separate issue for verifying if the context returned from different context sources are also filtered correctly. 

I was thinking since all the context will be turned into a context message, we can add do that during that process like we did for the interaction class. Do you think that'd work?

**Comment by DanTup** [comment]
> I was thinking since all the context will be turned into a context message, we can add do that during that process like we did for the interaction class. Do you think that'd work?

Sorry, missed this. I'm not sure I understand the question though - isn't the interaction class already filtering the context messages? I'm not very familiar with how the other context sources (like embeddings etc.) work, but I guess we should ensure that if any of those sources accidentally return ignored files, they get filtered out (not just going to the LLM, but in the UI so it doesn't look like they would be sent).

**Comment by DanTup** [comment]
nm - I saw https://github.com/sourcegraph/cody/issues/2911 / https://github.com/sourcegraph/cody/pull/2913 :)

**Comment by abeatrix** [comment]
> nm - I saw https://github.com/sourcegraph/cody/issues/2911 / https://github.com/sourcegraph/cody/pull/2913 :)

Haha yea! Just added you as a reviewer! Broke the e2e tests while I was trying to fix an issue on windows, so will need to fix that on Monday lol

**Comment by abeatrix** [comment]
@kalanchan are we good to close this out since all the initial items have been completed? 
I understand there are more works that the product team is planning on, and we can create a new issue for to track those when they are ready. Does this sound good to you?

**Comment by kalanchan** [comment]
@abeatrix yeah it seems like embeddings is the last thing to work on from this tracking issue? if so then its fine to close this as we wait for the next PRD to be approved and ready to go
